### PR TITLE
adds rule SnakeToCamelCasePropertiesAndVariablesRector

### PR DESCRIPTION
**Pull request by fforsthuber**

### CODE REVIEW

File: rules-tests/Naming/Rector/VarLikeIdentifier/SnakeToCamelCasePropertiesAndVariablesRector/Fixture/keep_constant.php.inc [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+?>
+-----
+<?php
+
+class SomeClass
+{
+    private const TEST_CONSTANT = 'test';
+}
+
+?>
```
Comment (samsonasik) [inline rules-tests/Naming/Rector/VarLikeIdentifier/SnakeToCamelCasePropertiesAndVariablesRector/Fixture/keep_constant.php.inc]: after part is not needed for skip fixture

```suggestion
```

**Comment by TomasVotruba** [comment]
@fforsthuber Hi, could you elaborate on this feature?

We had similar rules in the past. Those rule often affects external elements, e.g. public property rename and crashed. We'd like to see this as a community Rector rules set targeting naming conventions.

**Comment by fforsthuber** [comment]
Hi @TomasVotruba. You are right. The rule in this state would also change accesses of public properties of 'library' or 'vendor' classes/objects in your code base. I'll try to figure out how to prevent this.
Are there other known issues with such property renaming rules ?
By the way, thank you very much for coming up with such an awesome tool :)

File: rules-tests/Naming/Rector/VarLikeIdentifier/SnakeToCamelCasePropertiesAndVariablesRector/Fixture/keep_constant.php.inc (hunk 1 above)
Comment (fforsthuber) [reply rules-tests/Naming/Rector/VarLikeIdentifier/SnakeToCamelCasePropertiesAndVariablesRector/Fixture/keep_constant.php.inc]: Thanks :)

**Comment by samsonasik** [comment]
There are possible bugs that cause BC Break by this:

- may change also different objects same property
- change property used in parent/child/trait
- change non-private property ( even private property is accessible from parent public method)

Due to that, I think it should be in custom rule by your own use case.

**Comment by TomasVotruba** [comment]
> By the way, thank you very much for coming up with such an awesome tool :)

Glad you like it :)

---

> Are there other known issues with such property renaming rules ?

See the @samsonasik list. There is also troubles with autowiring, named aguments etc. Renaming is hard :)

The custom package focused on renames makes the most sense here :+1:

**Pull request closed without merging**